cachito/web/api_v1.py

```diff
@@ -104,7 +104,9 @@ def download_archive(request_id):
     flask.current_app.logger.debug(
         "Sending the bundle at %s for request %d", bundle_dir.bundle_archive_file, request_id
     )
-    return flask.send_file(str(bundle_dir.bundle_archive_file), mimetype="application/gzip")
+    return flask.send_file(
+        str(bundle_dir.bundle_archive_file), mimetype="application/gzip", as_attachment=True,
```

Comment:
Could you also set `attachment_filename` as requested in #86?

Reply:
@mprahl I tested the `send_file`, with `as_attachment=True`, the `Content-Disposition` header already has `filename` set with the same filename as the bundle archive file.

The doc tells:
> attachment_filename – the filename for the attachment if it **differs from** the file’s filename.

Reply:
@mprahl Are you thinking of to set a different filename rather than just the `{request.id}.tar.gz`?

Reply:
Got it. I didn't know that it was automatically set.

Optional: I'm thinking it might be nice to name it as `f"cachito-{request.id}.tar.gz"`.

Reply:
PR is modified to set this filename. Meanwhile, I also refactored the test without mocking flask.send_file, so that we can verify the expected result from response explicitly.